src/tokio.rs

```diff
@@ -337,6 +442,29 @@ where
     client_async_tls_with_connector_and_config(request, stream, connector, None).await
 }
 
+#[cfg(feature = "tokio-openssl")]
+/// Creates a WebSocket handshake from a request and a stream,
+/// upgrading the stream to TLS if required and using the given
+/// connector.
+pub async fn client_async_tls_with_connector<R, S>(
```

Comment:
This is the same as https://github.com/sdroege/async-tungstenite/blob/master/src/tokio.rs#L323-L327 or not? You could make the `#[cfg]` cover both cases, and also for the few functions above.

Reply:
This is actually slightly different, because the S type also has Sync and Send bounds, which is not the case for https://github.com/sdroege/async-tungstenite/blob/master/src/tokio.rs#L323-L327.

Reply:
Oh I see. Can we move that into a type alias like `AutoStream` so that the original functions can still be used instead of duplicating them just for the trait bounds?

Reply:
FWIW, if that's not possible then please duplicate also the other 2-3 functions above :) It's not like we don't have any duplicates of these elsewhere already because of all the `#[cfg]` mess, one more time is not going to hurt.

Reply:
I'm pretty sure there's no way for type aliases to be used for lists of trait bounds, so I'll duplicate the functions.

Reply:
I guess that's https://github.com/rust-lang/rust/issues/41517 :)